app/src/main/java/uiwidgetsc/listview/recyclerview/adapters/ExpandableListvieww.kt

```diff
@@ -0,0 +1,46 @@
+package uiwidgetsc.listview.recyclerview.adapters
+
+import android.graphics.Insets.add
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.example.kotlin_java_practicalss.R
+import com.example.kotlin_java_practicalss.databinding.ActivityExpandableListviewBinding
+import com.google.gson.Gson
+
+
+class ExpandableListvieww : AppCompatActivity() {
+
+    private lateinit var listviewAdapter: ExpandableListViewAdapter
+    private lateinit var movies: List<String>
+    private lateinit var movieDetails: HashMap<String, List<String>>
+    private lateinit var binding: ActivityExpandableListviewBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_expandable_listview)
+        binding = ActivityExpandableListviewBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        getList()
+        listviewAdapter = ExpandableListViewAdapter(this, movies, movieDetails)
+        binding.expandableListview.setAdapter(listviewAdapter)
+    }
+
+    private fun getList() {
+        movies = ArrayList()
+        movieDetails = HashMap()
+
+        val dClass = DataClass(this)
+        try {
+            (movies as ArrayList<String>).addAll(dClass.name)
```

Comment:
why do we need to do this all casting and in the try block?
if we use the proper model to bind data to the ExpandableListView then all this complexity would not be needed,
please fix this